src/index.js

```diff
@@ -61,13 +62,25 @@ const unvisit = (input, schema, getDenormalizedEntity) => {
   return schema.denormalize(input, unvisit, getDenormalizedEntity);
 };
 
+const getEntity = (entityOrId, schemaKey, entities) => {
+  if (typeof entityOrId === 'object') {
+    return entityOrId;
+  }
+
+  if (ImmutableUtils.isImmutable(entities)) {
```

Comment:
This could be sent in as a flag, since `entities` is a wrapped argument in the `getEntities` function. This would save on this conditional over every entity iteration. Something like...

```js
const getEntities = (entities, visitedEntities) => {
  const isImmutable = ImmutableUtils.isImmutable(entities);
  return (schema, entityOrId) => {
    // ...
    const entity = getEntity(entityOrId, schemaKey, entities, isImmutable);
    // ...
  };
};
```